toolbox/src/base/factory.cpp

```diff
@@ -36,7 +36,10 @@ wbt::Block* wbt::Block::instantiateBlockWithClassName(std::string blockClassName
 #ifdef WBT_USES_ICUB
     else if (blockClassName == wbt::MinimumJerkTrajectoryGenerator::ClassName) {
         block = new wbt::MinimumJerkTrajectoryGenerator();
-    } 
+    }
+    else if (blockClassName == wbt::DiscreteFilter::ClassName) {
```

Comment:
Oh sorry, now I see. Inside `WBT_USES_ICUB` is coherent, outside no. I fix both.